ios/browser/api/common/brave_foundation.h

```diff
@@ -0,0 +1,37 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_IOS_BROWSER_API_COMMON_BRAVE_FOUNDATION_H_
+#define BRAVE_IOS_BROWSER_API_COMMON_BRAVE_FOUNDATION_H_
+
+#import <Foundation/Foundation.h>
+
+#define OBJCPP_CF_OPTIONS(_type, _name)             \
+  enum __CF_OPTIONS_ATTRIBUTES _name : _type _name; \
+  enum _name : _type
+#define OBJCPP_NS_OPTIONS(_type, _name) OBJCPP_CF_OPTIONS(_type, _name)
+
+#if defined(__cplusplus)
+#define OBJCPP_TO_UNDERLYING_TYPE(value) \
+  static_cast<typename std::underlying_type<decltype(value)>::type>(value)
+
+#define OBJCPP_NS_OPTIONS_IMPLEMENT_OPERATORS(_type)             \
+  inline _type operator|(_type left, _type right) {              \
+    return static_cast<_type>(OBJCPP_TO_UNDERLYING_TYPE(left) |  \
+                              OBJCPP_TO_UNDERLYING_TYPE(right)); \
+  }                                                              \
+  inline _type operator&(_type left, _type right) {              \
+    return static_cast<_type>(OBJCPP_TO_UNDERLYING_TYPE(left) &  \
+                              OBJCPP_TO_UNDERLYING_TYPE(right)); \
+  }                                                              \
+  inline _type& operator|=(_type& left, _type right) {           \
+    return left = left | right;                                  \
+  }                                                              \
+  inline _type& operator&=(_type& left, _type right) {           \
+    return left = left | right;                                  \
+  }
+#endif
```

Comment:
This seems like a bit much just to forward declare enum options, still think personally its just better if we include enum header in all places where its needed